utils/gen_artifacts_toml.jl

```diff
@@ -20,16 +19,34 @@ for format ∈ formats
   global k
   for matrix ∈ eachrow(matrices)
     k += 1
-    url = "$ssmc_url/$format/$(matrix.group)/$(matrix.name).tar.gz"
+    if format == "mat"
+      sscm_file = "$(matrix.name).mat"
+      url = "$ssmc_url/$format/$(matrix.group)/$sscm_file"
+    else
+      sscm_file = "$(matrix.name).tar.gz"
+      url = "$ssmc_url/$format/$(matrix.group)/$sscm_file"
+    end
     println("$k/$nmatrices: ", url)
     try
-      add_artifact!(
-        artifacts_toml,
-        "$(matrix.group)/$(matrix.name).$(format)",
-        url,
-        lazy = true,
-        force = true,
-      )
+      problem_hash = artifact_hash("$(matrix.group)/$(matrix.name).$(format)", artifacts_toml)
+      if problem_hash === nothing || !artifact_exists(problem_hash)
+        problem_hash = create_artifact() do artifact_dir
+          download("$url", joinpath(artifact_dir, sscm_file))
+        end
```

Comment:
@AntoninKns
Do you if we should generate a `git-tree-sha1` like this or with https://github.com/simeonschaub/ArtifactUtils.jl/blob/main/src/ArtifactUtils.jl#L52-L57 ?